book_maker/loader/srt_loader.py

```diff
@@ -1 +1,325 @@
+
+        try:
+            with open(f"{self.srt_name}", encoding="utf-8") as f:
+                self.blocks = self._parse_srt(f.read())
+        except Exception as e:
+            raise Exception("can not load file") from e
+
+        index = 0
+        p_to_save_len = len(self.p_to_save)
+
+        try:
+            sliced_list = self._get_sliced_list()
+
+            for sliced in sliced_list:
+                begin, end, text = sliced
+
+                if not self.resume or index + (end - begin) > p_to_save_len:
+                    if index < p_to_save_len:
+                        self.p_to_save = self.p_to_save[:index]
+
+                    try:
+                        temp = self.translate_model.translate(text)
+                    except Exception as e:
+                        print(e)
+                        raise Exception("Something is wrong when translate") from e
+
+                    translated_blocks = self._get_blocks_from(temp)
+
+                    if self.accumulated_num > 1:
+                        if not self._check_blocks(
+                            translated_blocks, self.blocks[begin:end]
+                        ):
+                            translated_blocks = []
+                            # try to translate one by one, so don't accumulate too much
+                            print(
+                                f"retry it one by one:  {self.blocks[begin]['number']} - {self.blocks[end-1]['number']}"
+                            )
+                            for block in self.blocks[begin:end]:
+                                try:
+                                    temp = self.translate_model.translate(
+                                        self._get_block_translate(block)
+                                    )
+                                except Exception as e:
+                                    print(e)
+                                    raise Exception(
+                                        "Something is wrong when translate"
+                                    ) from e
+                                translated_blocks.append(self._get_block_from(temp))
+
+                            if not self._check_blocks(
+                                translated_blocks, self.blocks[begin:end]
+                            ):
+                                raise Exception(
+                                    f"retry failed, adjust the srt manually."
+                                )
+
+                    i = 0
+                    for block in translated_blocks:
+                        text = block.get("text", "")
+                        self.p_to_save.append(text)
+                        self.bilingual_result.append(
+                            f"{self._get_block_text(self.blocks[begin + i])}\n{text}"
+                        )
+                        i += 1
+                else:
+                    i = 0
+                    for block in self.blocks[begin:end]:
```

Comment:
ditto

Reply:
Let me fix this, not quite familiar with the python way. 😓